Allows multiple flush on the same file

guj opened the pull request:
This is to resolve [WarpX issue 561](https://github.com/ECP-WarpX/WarpX/issues/561), which flush is called more than once.  And the later times not all processes participated while H5Fopen is a collective call.

guj committed:
This is to resolve issue 561, which flush is called more than once.
And the later times not all processes participated while H5Fopen is
a collective call

ax3l commented on src/IO/HDF5/HDF5IOHandler.cpp:
```diff
@@ -352,6 +352,11 @@ HDF5IOHandlerImpl::openFile(Writable* writable,
     if( !auxiliary::ends_with(name, ".h5") )
         name += ".h5";
 
+    auto search = m_fileIDs.find(writable);
```
This is effectively implementing the `//TODO` above, isn't it?
Maybe we can add this as a comment before this block?

guj replied:
Yes,  looks like so.  Well  CI failed.  let me address that first

guj committed:
added a map function to  collect hdf5  name/id pairs
to avoid duplicationa and wrong id assignment
when new iteration file is created

guj commented:
@ax3l  Hi Axel,  please review and merge  it when you get a chance.  Thanks.

ax3l reviewed (approved):
Thank you, this looks great to me!
Did this already address the issue in WarpX or do you need to add similar support to the iteration, as mentioned in the TODO comment?
